src/model/resources/flexible_ccs/ccs_solvent_storage.jl

```diff
@@ -0,0 +1,389 @@
+    @expression(EP, eCFom_CCS_SS[y in CCS_SOLVENT_STORAGE, i in 1:7],
+        solvent_storage_dict[y,"fom_cost"][i] * eTotalCap_CCS_SS[y,i])
+
+    # Total fixed cost expression - combines investment and fixed O&M costs
+    @expression(EP, eCFix_CCS_SS[y in CCS_SOLVENT_STORAGE, i in 1:7],
+        if y in NEW_CAP_CCS_SS
+            # For resources with new capacity: investment cost + fixed O&M cost
+            EP[:eCInv_CCS_SS][y, i] + EP[:eCFom_CCS_SS][y, i]
+        else
+            # For existing resources: only fixed O&M cost
+            EP[:eCFom_CCS_SS][y, i]
+        end)
+
+    # connect eCFix_CCS_SS_Plant to eCFix
+    @expression(EP, eCFix_CCS_SS_Plant[y in CCS_SOLVENT_STORAGE], sum(EP[:eCFix_CCS_SS][y,i] for i in 1:7))
+    @expression(EP, eTotalCFix_CCS_SS, sum(EP[:eCFix_CCS_SS_Plant][y] for y in CCS_SOLVENT_STORAGE))
+    
+    # connect eCInv_CCS_SS_Plant to eCInv
+    @expression(EP, eCInv_CCS_SS_Plant[y in CCS_SOLVENT_STORAGE], sum(EP[:eCInv_CCS_SS][y,i] for i in 1:7))
+    @expression(EP, eTotalCInv_CCS_SS, sum(EP[:eCInv_CCS_SS_Plant][y] for y in CCS_SOLVENT_STORAGE))
+    
+    # connect eCFom_CCS_SS_Plant to eCFom
+    @expression(EP, eCFom_CCS_SS_Plant[y in CCS_SOLVENT_STORAGE], sum(EP[:eCFom_CCS_SS][y,i] for i in 1:7))
+    @expression(EP, eTotalCFom_CCS_SS, sum(EP[:eCFom_CCS_SS_Plant][y] for y in CCS_SOLVENT_STORAGE))
+    
+    # add this to eTotalCFix
+    add_to_expression!(EP[:eTotalCFix], eTotalCFix_CCS_SS)
+    # add this to eTotalCInv
+    add_to_expression!(EP[:eTotalCInv], eTotalCInv_CCS_SS)
+    # add this to eTotalCFom
+    add_to_expression!(EP[:eTotalCFom], eTotalCFom_CCS_SS)
+
+    # add to Obj
+    if MultiStage == 1
+        # OPEX multiplier scales fixed costs to account for multiple years between two model stages
+        # We divide by OPEXMULT since we are going to multiply the entire objective function by this term later,
+        # and we have already accounted for multiple years between stages for fixed costs.
+        add_to_expression!(EP[:eObj], 1 / inputs["OPEXMULT"], eTotalCFix_CCS_SS)
+    else
+        add_to_expression!(EP[:eObj], eTotalCFix_CCS_SS)
+    end
+
+    if MultiStage == 1
+        # Existing capacity variable is equal to existing capacity specified in the input file
+        @constraint(EP, cExistingCap_CCS_SS[y in CCS_SOLVENT_STORAGE, i in 1:7], EP[:vEXISTINGCAP_CCS_SS][y,i]== solvent_storage_dict[y, "existing_cap"][i])
+    end
+
+    # add vom
+    # variable OM 
+    @expression(EP, eCVar_CCS_SS_unit[y in CCS_SOLVENT_STORAGE, i = 1:5, t = 1:T], omega[t] * vOutput_CCS_SS[y,i,t] * solvent_storage_dict[y,"vom_cost"][i])
+    # sum to annual level 
+    @expression(EP, eCVar_CCS_SS[y in CCS_SOLVENT_STORAGE], sum(EP[:eCVar_CCS_SS_unit][y,i,t] for i in 1:5 for t in 1:T))
+    # sum to zonal-annual level
+    @expression(EP, eZonalCVar_CCS_SS[z = 1:Z], sum(EP[:eCVar_CCS_SS][y] for y in intersect(CCS_SOLVENT_STORAGE, resources_in_zone_by_rid(gen, z))))
```

Comment:
Pre-compute set outside @expression

```suggestion
    CCS_SOLVENT_STORAGE_BY_ZONE = map(1:Z) do z
        return intersect(CCS_SOLVENT_STORAGE, resources_in_zone_by_rid(gen, z))
    end
    @expression(EP, eZonalCVar_CCS_SS[z = 1:Z], 
        sum(EP[:eCVar_CCS_SS][y] 
        for y in CCS_SOLVENT_STORAGE_BY_ZONE[z]))
```